app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataActivity.kt

```diff
@@ -80,8 +82,35 @@ class MetadataActivity : AppCompatActivity() , MetadataView, CaptureFlow {
         Toast.makeText(this, R.string.metadata_invalid_form, Toast.LENGTH_SHORT).show()
     }
 
-    override fun finishFlow() {
-        goToHome()
+    private fun save() {
+        coroutineScope.launch {
+            try {
+                presenter.save(
+                    metadata_section_smear_type_radio_group.checkedRadioButtonId,
+                    metadata_species_spinner.selectedItem.toString(),
+                    metadata_stage_spinner.selectedItem.toString()
+                )
+                finishFlow()
```

Comment:
I don´t understand why you have move the method from the presenter to the activity, this is not the normal way to proceed, normally you need to execute the logic in the presenter and show the result in the activity. Also if you need to catch any error you need to do it in the presenter and have a method to success and a method to error that calls the activity.... i don´t understand to much the reason to do in this way, is possible to change?

Reply:
Thanks, I'll have a look!

Reply:
Modified attending your suggestion